engines/e_pkcs11.c

```diff
@@ -223,14 +226,36 @@ static CK_SLOT_ID pkcs11_get_slot()
         goto err;
     }
 
-    /* TODO: make mechanism to provide other slotid */
+    slotList = OPENSSL_malloc(sizeof(CK_SLOT_ID) * slotCount);
+
+    if (slotList == NULL) {
+        PKCS11err(PKCS11_F_PKCS11_GET_SLOT, PKCS11_R_MEMORY_ALLOCATION_FAILED);
+        OPENSSL_free(slotList);
```

Comment:
no need to free a NULL pointer.

Reply:
Thanks @FdaSilvaYY, nice to meet you here too